### PR TITLE
Add a golbal trace provider

### PR DESCRIPTION
**Pull request by lengrongfu**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

Where used call otel.GetTracerProvider() to get this provider.

#### Which issue(s) this PR fixes:

Fixes # https://github.com/kubernetes/kubernetes/issues/127705

#### Special notes for your reviewer:

We hope in newTracerProvider method add otel.SetTracerProvider(tp), to set global trace provider.

Test pass:
image

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubemark/hollow_kubelet.go [hunk 1]
```diff
@@ -111,7 +110,6 @@ func NewHollowKubelet(
 		HostUtil:                  hostutil.NewFakeHostUtil(nil),
 		PodStartupLatencyTracker:  kubeletutil.NewPodStartupLatencyTracker(),
 		NodeStartupLatencyTracker: kubeletutil.NewNodeStartupLatencyTracker(),
-		TracerProvider:            noopoteltrace.NewTracerProvider(),
```
Comment (googs1025) [inline pkg/kubemark/hollow_kubelet.go:114]: do we have to remove this?

Comment (lengrongfu) [reply pkg/kubemark/hollow_kubelet.go:114]: yes, but we need add a global TracerProvider.

**Comment by dims** [comment]
cc @dashpole

**Comment by dashpole** [comment]
While this is simpler, we've avoided using global registries for metrics because it can cause issues: https://github.com/kubernetes/kubernetes/issues/75779.  I think we should continue to avoid globals for traces as well.

**Comment by lengrongfu** [comment]
Due to past problems, no merger

**Pull request closed without merging**